ocs_ci/helpers/sanity_helpers.py

```diff
@@ -183,3 +182,133 @@ def __init__(self):
         self.pod_objs = list()
         self.obc_objs = list()
         self.ceph_cluster = CephClusterExternal()
+
+
+class SanityManagedService(Sanity):
+    """
+    Class for cluster health and functional validations for the Managed Service
+    """
+
+    def __init__(
+        self,
+        create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers,
+        scale_count=None,
+        pvc_per_pod_count=5,
+        start_io=True,
+        io_runtime=None,
+        pvc_size=None,
+        max_pvc_size=30,
+        consumer_indexes=None,
+    ):
+        """
+        Init the sanity managed service class.
+        Init the 'create resources on MS consumers' factory.
+        This method uses the 'create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers' factory.
+
+        Args:
+           create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers (function): Factory for creating scale
+               pods and PVCs using k8s on MS consumers fixture.
+           scale_count (int): No of PVCs to be Scaled. Should be one of the values in the dict
+               "constants.SCALE_PVC_ROUND_UP_VALUE".
+           pvc_per_pod_count (int): Number of PVCs to be attached to single POD
+               Example, If 20 then 20 PVCs will be attached to single POD
+           start_io (bool): Binary value to start IO default it's True
+           io_runtime (seconds): Runtime in Seconds to continue IO
+           pvc_size (int): Size of PVC to be created
+           max_pvc_size (int): The max size of the pvc
+           consumer_indexes (list): The list of the consumer indexes to create scale pods and PVCs.
+               If not specified - if it's a consumer cluster, it creates scale pods and PVCs only
+               on the current consumer. And if it's a provider it creates scale pods and PVCs on
+               all the consumers.
+        """
+        super(SanityManagedService, self).__init__()
+        # A dictionary of a consumer index per the fio_scale object
+        self.consumer_i_per_fio_scale = {}
+
+        self.create_resources_on_ms_consumers_factory = (
+            create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers
+        )
+        self.scale_count = scale_count
+        self.pvc_per_pod_count = pvc_per_pod_count
+        self.start_io = start_io
+        self.io_runtime = io_runtime
+        self.pvc_size = pvc_size
+        self.max_pvc_size = max_pvc_size
+
+        if consumer_indexes:
+            self.consumer_indexes = consumer_indexes
+        elif is_ms_consumer_cluster():
```

Comment:
If the current cluster is consumer then only the current cluster index is saved. Why are not stored also indexes of other consumers at this step?

Reply:
The assumption is that if the cluster is a consumer, we will prefer to create resources only for the consumer. And if it's a provider, we like to create resources for all the consumers.

Reply:
This sounds inconsistent and complicate to me. In all other cases all indexes are stored in `self.consumer_indexes` variable and later used but in one case it is not stored depending on where the test is executed. I would prefer if this was consistent and the check was triggered on all available consumers no matter where it is executed. If we want an option to check health on one cluster then I would suggest to use only the `consumer_indexes` parameter of the function as you already have defined.

Reply:
After our discussion, I agree that your approach is ok. In most cases we don't have to check all consumers. Please just add more log messages into the code to make it easier to identify what indexes were passed and what should be checked.

Reply:
Okay, np.